lib/services/service_brokers/v2/catalog_schemas.rb

```diff
@@ -1,10 +1,13 @@
+require 'json-schema'
+
 module VCAP::Services::ServiceBrokers::V2
+  MAX_SCHEMA_SIZE = 65_536
   class CatalogSchemas
     attr_reader :errors, :create_instance
 
-    def initialize(attrs)
+    def initialize(schema)
       @errors = VCAP::Services::ValidationErrors.new
-      validate_and_populate_create_instance(attrs)
+      validate_and_populate_create_instance(schema)
```

Comment:
(This comment doesn't necessarily reflect changes form this specific PR, but the set of PRs)

We were surprised to see validations happening in the constructor. We are more familiar with the ActiveModel pattern of not explicitly implementing `valid?`, but rather including [`validate`](http://guides.rubyonrails.org/active_record_validations.html) statements.

With this approach, we'd be able to convert the basic validations (data types, presence checks, length) with the validations provided by ActiveModel, and separate those from the content validations.

We're asking for this change to make the code consistent with recent V3 code where validations are important. Examples are in the [`messages/`](https://github.com/cloudfoundry/cloud_controller_ng/tree/master/app/messages) folder